File: Clients/BlogBlazor/Shared/Blazor.Contracts/V1/Validators/MessagesValidators/CreateMessageRequestValidator.cs
using Blazor.Contracts.V1.Requests.MessagesRequests;
using FluentValidation;

namespace Blazor.Contracts.V1.Validators.MessagesValidators;

/// <summary>
/// Create message request validator.
/// </summary>
/// <seealso cref="AbstractValidator{CreateMessageRequest}" />
public class CreateMessageRequestValidator : AbstractValidator<CreateMessageRequest>
{
    /// <summary>
    /// Initializes a new instance of the <see cref="CreateMessageRequestValidator"/> class.
    /// </summary>
    public CreateMessageRequestValidator()
    {
        RuleFor(x => x.Subject)
            .NotEmpty()
            .Matches("^[a-zA-Z0-9 ]*S");

        RuleFor(x => x.Body)
            .NotEmpty()
            .Matches("^[a-zA-Z0-9 ]*S");
    }
}

File: Shared/Blog.FSharp.Contracts/V1/Validators/MessagesValidators/CreateMessageRequestValidator.fs
namespace Blog.FSharp.Contracts.V1.Validators.MessagesValidators

open FluentValidation
open Blog.FSharp.Contracts.V1.Requests.MessagesRequests

/// <summary>
/// Validator for creating a message request.
/// </summary>
type CreateMessageRequestValidator() =
    inherit AbstractValidator<CreateMessageRequest>()

    do
        // Ensure Subject is not empty and matches the regex
        base.RuleFor(fun x -> x.Subject)
            .NotEmpty()
            .Matches("^[a-zA-Z0-9 ]*S") |> ignore

        // Ensure Body is not empty and matches the regex
        base.RuleFor(fun x -> x.Body)
            .NotEmpty()
            .Matches("^[a-zA-Z0-9 ]*S") |> ignore

